tests/console/python3_websocket_client.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: python3-websocket-client tests
+# - ensure python3 module is enabled in the system
+# - install python-websocket-client package
+# - use library to connect and transfer some data to a test server
+# - Compare the result vs expected
+#
+# Maintainer: QE-Core <qe-core@suse.de>
+
+use base 'consoletest';
+use warnings;
+use strict;
+use v5.20;
+use feature qw(signatures);
+no warnings qw(experimental::signatures);
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils 'zypper_call';
+use python_version_utils;
+use version_utils 'is_sle';
+use registration;
+
+sub test_setup {
+    select_serial_terminal;
+    # Import python scripts
+    assert_script_run("curl -O " . data_url("python/websockets/client-test.py"));
+    assert_script_run("curl -O " . data_url("python/websockets/server.py"));
+    # install server dependencies
+    zypper_call("install python3-tornado");
+    # start websocket server in background
+    return background_script_run 'python3 server.py';
+}
+
+sub run_test ($python_package) {
+    return unless script_run("zypper search $python_package-websocket-client") == 0;
+    record_info("Testing for $python_package");
+    zypper_call("install $python_package $python_package-websocket-client");
+    my $python_interpreter = get_python3_binary($python_package);
+    record_info("running python version", script_output("$python_interpreter --version"));
+    # Execute python script. The script itself ensure output is the one expected
+    assert_script_run("$python_interpreter client-test.py");
+    # clean up for the next run
+    zypper_call("rm $python_package $python_package-websocket-client");
```

Comment:
good point; we did the same for other python tests, so basically we treat the "system Python" as any other version and make the test independent: each test installs ALL the available pythons and de-install them to clean after finish. My suggestion is to not rely on the concept of having a "system python" at all since it can change on any update. WDYT ?

Reply:
And if any subsequent test fails because it expects to have a Python installed by default, that need to be fixed. IMO Tests should be atomic/independent and make no assumptions on the existing environment :)